test/instance_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/environment.h"
+#include "../src/instance.h"
+#include "../src/resource.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+namespace {
+
+class InstanceTest : public ::testing::Test {
+ protected:
+  const MetadataStore::Metadata&& GetResourceMetadata(
```

Comment:
How about:
```cpp
  const MetadataStore::Metadata& GetMetadata(
      const MetadataUpdater::ResourceMetadata& rm) {
    return rm->metadata;
  }
```
? You don't actually need the ownership, you just want to get a reference to compare with.